nats/config.go

```diff
@@ -0,0 +1,97 @@
+package nats
+
+import (
+	"context"
+	"log/slog"
+	"os"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+// Config defines the config for storage.
+type Config struct {
+	// Nats URL, default "nats://127.0.0.1:4222"
+	URL string
+	// Nats username
+	Username string
+	// Nats password
+	Password string
+	// Nats credentials file: https://docs.nats.io/using-nats/developer/connecting/creds
+	CredentialsFile string
+	// Nats client name
+	ClientName string
+	// Nats retry on failed connect: https://docs.nats.io/using-nats/developer/connecting/reconnect
+	RetryOnFailedConnect bool
+	// Nats max reconnects: https://docs.nats.io/using-nats/developer/connecting/reconnect
+	MaxReconnects int
+	// Nats context
+	Context context.Context
+	// Nats key value config
+	KeyValueConfig jetstream.KeyValueConfig
+	Logger         *slog.Logger
+	// Applicable only if Logger is nil.
+	// Until go 1.22, it is weird to set log level.
+	// See https://github.com/golang/go/issues/62418
+	LogLevel slog.Level
```

Comment:
Can we do this without slog? So that we can support 1.20?

Reply:
You are right, I should not force go upgrade just because of logging. I just found the standard log ..lacking.

Reply:
What do you say about using something like zerolog instead?

Reply:
Either Fiber Logger or Fiber Zerolog from contrib: https://github.com/gofiber/contrib/tree/main/fiberzerolog